### PR TITLE
TA125917 TA125849 Making sure the test renders the dynamic elements

### PR DESCRIPTION
**Pull request by lawong-d2l**
https://rally1.rallydev.com/#/15545269080ud/detail/task/226740798392
https://rally1.rallydev.com/#/15545269080ud/detail/task/227292092924

### CODE REVIEW

File: test/d2l-lesson-header.html [hunk 1]
```diff
@@ -15,46 +15,57 @@
...
+							return res.json();
+						})
+						.then(function(entities) {
+							element = fixture('LessonFixture');
+							element.href = 'data/lesson1.json';
+							element.entity = window.D2L.Hypermedia.Siren.Parse(entities);
+							const dynamicElements = element.shadowRoot.querySelectorAll('dom-if');
+							if (dynamicElements) {
+								for (let i = 0; i < dynamicElements.length; i++) {
+									dynamicElements[i].render();
```
Comment (jwong-d2l) [inline test/d2l-lesson-header.html]: What does rendering all the dom-ifs do?

Comment (RyanSweetD2L) [reply test/d2l-lesson-header.html]: It prevents us having to `flush()` or `setTimeout()` because it realizes all the content.

Comment (lawong-d2l) [reply test/d2l-lesson-header.html]: It's been recommended to me on the Polymer slack testing channel, as well as some people in the office. Polymer's own tests even use it to make sure it gets stamped. https://github.com/Polymer/polymer/blob/master/test/unit/dom-if.html#L141

File: test/d2l-lesson-header.html [hunk 2]
```diff
@@ -15,46 +15,57 @@
...
 
-				test('should create module title element', function(done) {
-					flush( function() {
-						var moduleTitle = element.shadowRoot.querySelector('.module-title');
-						expect( moduleTitle ).to.exist;
-						expect( moduleTitle.innerHTML === 'Lesson 1' );
-						done();
-					});
+				test('should create module title element', function() {
+					const moduleTitle = element.shadowRoot.querySelector('.module-title');
```
[the remark this answers is not in the document]
Comment (lawong-d2l) [reply test/d2l-lesson-header.html]: A const variable is never reassigned.

[the remark this answers is not in the document]
Comment (RyanSweetD2L) [reply test/d2l-lesson-header.html]: Prefer `const`.  Use let only when you have to :)

File: test/d2l-lesson-header.html [hunk 3]
```diff
@@ -15,46 +15,57 @@
...
 		</test-fixture>
-
 		<script>
-
 			suite('d2l-lesson-header for a lesson', () => {
 
-				var element = fixture('LessonFixture');
+				let element;
+
+				setup(function(done) {
```
[the remark this answers is not in the document]
Comment (lawong-d2l) [reply test/d2l-lesson-header.html]: I don't think so. I did this is to actually help with the flakiness more. I'm not sure why the test were wrote this way. I'd rather keep them consistent to what Eric showed me on how to set them up.

**Pull request merged**